packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -211,3 +354,240 @@ export async function provideTspconfigCompletionItems(
     return distinctArray(result, (t) => t.label);
   }
 }
+
+/**
+ * Get the new text value
+ * @param sourceQuoteType input quote type(single, double, none)
+ * @param formatText format text value
+ * @returns new text value
+ */
+function getCompletionItemInsertedValue(sourceQuoteType: string, formatText: string): string {
+  let newText: string = "";
+  if (sourceQuoteType === "QUOTE_SINGLE") {
+    newText = `${formatText}'`;
+  } else if (sourceQuoteType === "QUOTE_DOUBLE") {
+    newText = `${formatText}"`;
+  } else {
+    newText = `"${formatText}"`;
+  }
+  return newText;
+}
+
+/**
+ * Get the full completion item value and edit position
+ * @param newText  the new text
+ * @param source source text
+ * @param nodePosRange current node position range, [startPos, endPos, nodeEndPos]
+ * @param curPos current cursor position
+ * @param tspConfigPosition original position, see {@link Position}
+ * @returns TextEdit object or undefined
+ */
+function getFullCompletionItemValAndEditPosition(
+  newText: string,
+  source: string,
+  nodePosRange: number[],
+  curPos: number,
+  tspConfigPosition: Position,
+): TextEdit | undefined {
+  const [startPos, endPos] = nodePosRange;
+
+  if (curPos >= startPos && curPos <= endPos) {
+    return TextEdit.replace(
+      Range.create(
+        Position.create(tspConfigPosition.line, tspConfigPosition.character - source.length),
+        Position.create(tspConfigPosition.line, tspConfigPosition.character + (endPos - curPos)),
+      ),
+      newText,
+    );
+  }
+  return undefined;
+}
+
+/**
+ * Get the common CompletionItem object, which value is included in quotes
+ * @param labelName The value of the label attribute of the CompletionItem object
+ * @param sourceType Input quote type
+ * @param source Entered text
+ * @param description The value of the documentation attribute of the CompletionItem object
+ * @param tspConfigPosition Input location object, see {@link Position}
+ * @param nodePosRange current node position range, [startPos, endPos, nodeEndPos]
+ * @param curPos current cursor position
+ * @returns CompletionItem object or undefined
+ */
+function createContainingQuatedValCompetionItem(
+  labelName: string,
+  sourceType: string,
+  source: string,
+  description: string,
+  tspConfigPosition: Position,
+  nodePosRange: number[],
+  curPos: number,
+): CompletionItem | undefined {
+  // Generate new text
+  const newText = getCompletionItemInsertedValue(sourceType, labelName);
+  const textEdit = getFullCompletionItemValAndEditPosition(
```

Comment:
feels lamda is better here considering you are passing lots of params and the function is small